gpcontrib/pxf_fdw/pxf_fragment.c

```diff
@@ -5,7 +5,7 @@
 #include "cdb/cdbtm.h"
 #include "cdb/cdbvars.h"
 #include "commands/copy.h"
-#if (PG_VERSION_NUM <= 140000)
+#if (PG_VERSION_NUM >= 140000)
```

Comment:
Should be <= 150000?

Reply:
I'm not quite sure.  Current PG version in CloudberryDB is 14.4, PG_VERSION_NUM is 140004.

Maybe we see what's going on at that time we bring in PG15?

This change is just it fixed this issue and it kept same way as code in pxf_fdw.c

#if (PG_VERSION_NUM >= 140000)
#include "commands/copyfrom_internal.h"
#include "commands/copyto_internal.h"
#endif

Reply:
I am very confused how PG_VERSION_NUM is assigned. In configure.ac (see diff below), I changed the Cloudberry version to 1.6.3 and change PG_PACKAGE_VESION to 15.4.

After running `configure`, notice the value (`150003`) assigned to `PG_VERSION_NUM`. It seems as if it takes the 15 from PG_PACKAGE_VESION and the trailing two digits `03` from the last digit of the Cloudberry Database version:

```
src/include/pg_config.h:944:#define PG_VERSION_NUM 150003
```

Here is the only change I made to configure.ac:
```
diff --git a/configure.ac b/configure.ac
index 71e5e3c..1fe2c5c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,8 +20,8 @@ m4_pattern_forbid(^PGAC_)dnl to catch undefined macros
 dnl The PACKAGE_VERSION from upstream PostgreSQL is maintained in the
 dnl PG_PACKAGE_VERSION variable, when merging make sure to update this
 dnl variable with the merge conflict from the AC_INIT() statement.
-AC_INIT([Cloudberry Database], [1.0.0], [info@cloudberrydb.org], [], [https://www.cloudberrydb.org/])
-[PG_PACKAGE_VERSION=14.4]
+AC_INIT([Cloudberry Database], [1.6.3], [info@cloudberrydb.org], [], [https://www.cloudberrydb.org/])
+[PG_PACKAGE_VERSION=15.4]
 AC_SUBST(PG_PACKAGE_VERSION)

 dnl m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.69], [], [m4_fatal([Autoconf version 2.69 is required.
 ```

I believe your fix does address the compilation issue, but is there something fundamentally wrong in how PG_VERSION_NUM is assigned a value?

Reply:
For above strange behavior  you described how PG_VERSION_NUM is assigned after change , I agreed it's strange for me too,  but I have no idea.

Reply:
first of all, we do not need parentheses here
second, `commands/copyfrom_internal.h` was committed as part of https://github.com/postgres/postgres/commit/c532d15dddff14b01fe9ef1d465013cb8ef186df and still is part of PostgreSQL 
So, condition should be 
`#if PG_VERSION_NUM >= 140000`
As for PG_VERSION_NUM its value should come from PostgreSQL core major and minor version. So, given Cloudberry is based on 14.4 PG_VERSION_NUM should be equal to 140004
I didnt take a look on Cloudburry version generation, but if it results in anything different from 140004 then we have a bug here.

Reply:
@reshke Thank you for confirmation. As it stands today, the last digit of the Cloudberry Database version is used as the last digit assigned to PG_VERSION_NUM.  We have a bug.